modules/packages/LinearAlgebra.chpl

```diff
@@ -1306,16 +1295,14 @@ proc lu (A: [?Adom] ?eltType) {
   if Adom.rank != 2 then
     halt("Wrong rank for LU factorization");
 
-  if Adom.shape(0) != Adom.shape(1) then
-    halt("LU factorization only supports square matrices");
 
   var (LU, ipiv, numSwap) = _lu(A);
   return (LU,ipiv);
 }
 
 /* Return a new array as the permuted form of `A` according to
     permutation array `ipiv`.*/
-private proc permute (ipiv: [] int, A: [?Adom] ?eltType, transpose=false) {
+proc permute (ipiv: [] int, A: [?Adom] ?eltType, transpose=false) {
```

Comment:
Was this change based on some discussion somewhere? I am OK with it, but when we make something public, we try to do a survey of similar functions on other similar libraries. And I can't tell by just looking at this PR whether this change has gone through that process.

Reply:
Hi!
Sorry for changing the method modifier without consulting. I feel that after an LU decomposition, one would need to permute the matrix using the permutation array returned from the `lu` procedure for any further operations. This includes recovering the original matrix.(which is done in the test files)